### PR TITLE
[ROCM] Add plumbing for hlo_to_llvm_ir tests

### PR DESCRIPTION
**Pull request by draganmladjenovic**

### CODE REVIEW

**Comment by i-chaochen** [comment]
@ddunl will you have a look on this, please? Thanks

**Comment by draganmladjenovic** [comment]
For context. Previous discussion is at https://github.com/tensorflow/tensorflow/pull/58441.

**Comment by ddunl** [comment]
I think @akuegel is probably better suited to look at this one. Looks like I need to make my check_contents.py a bit more robust too

**Review by akuegel — APPROVED** [review]

**Comment by sherhut** [comment]
@ddunl can you look into why the check content script is failing?

File: xla/runlit.cfg.py [hunk 1]
```diff
@@ -57,6 +57,19 @@
 else:
   llvm_config.use_default_substitutions()
 
+subst_marker = "SUBST_";
```
Comment (akuegel) [inline xla/runlit.cfg.py:60]: I see some failures in a Kokoro job for testing OSS GPU, it looks like this config is not actually being applied. Is this actually working on your end?
Internally I got it working by adjusting our config we use internally to do the replacements. But it seems it doesn't work for open source.

Comment (akuegel) [reply xla/runlit.cfg.py:60]: Nevermind, I think I found the problem, tensorflow/compiler/mlir/runlit.cfg.py also needs to be adjusted, it is used for tensorflow testing.

Comment (draganmladjenovic) [reply xla/runlit.cfg.py:60]: @akuegel Thanks! Do I need to send a separate PR to the tensoflow repo?

Comment (akuegel) [reply xla/runlit.cfg.py:60]: I am taking care of it.

**Pull request closed without merging**